### PR TITLE
test: removed the usage of default_configuration.

### PR DESCRIPTION
**Pull request by NickNaso**
This PR remove the usage of  `process.config.target_defaults.default_configuration` in all test suite.
A new function `whichBuildType` has been introduced and it has the reponsability to check which type of build `node-gyp` made so the test can refer to the right folders.

Refs:
- https://github.com/nodejs/node-addon-api/issues/1207

### CODE REVIEW

File: test/common/index.js [hunk 1]
```diff
@@ -75,9 +76,35 @@ exports.mustNotCall = function (msg) {
...
+async function checkBuildType (buildType) {
+  try {
+    await access(path.join(path.resolve('./test/build'), buildType));
+    return true;
+  } catch {
+    return false;
+  }
+}
 
+async function whichBuildType () {
```
Comment (legendecas) [inline test/common/index.js:93]: What about determining with a simple environ like `BUILD_TYPE`?

Comment (legendecas) [reply test/common/index.js:93]: Note that if people incremental build with `node-gyp build -C test --debug`, the build directory will not be cleaned and `build/Release` can still present.

File: test/common/index.js [hunk 2]
```diff
@@ -91,12 +91,25 @@ async function checkBuildType (buildType) {
...
 
 async function whichBuildType () {
-  let buildType = 'Release';
-  if (await checkBuildType(buildTypes.Release)) {
-    buildType = buildTypes.Release;
-  }
-  if (await checkBuildType(buildTypes.Debug)) {
-    buildType = buildTypes.Debug;
+  let buildType;
+  const envBuildType = process.env.BUILD_TYPE;
```
Comment (mhdawson) [inline test/common/index.js:95]:
```suggestion
  const envBuildType = process.env.BUILD_TYPE || 'Release';
```

Comment (mhdawson) [reply test/common/index.js:95]: Suggestion in meeting from @vmoroz how about `process.env.NODE_API_BUILD_CONFIG`

**Review by legendecas — APPROVED** [review]

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by mhdawson** [comment]
Landed as 10ad762807925c88109e45db728bc0ed87153a2d

**Pull request closed without merging**